src/adaptation/icap/Xaction.cc

```diff
@@ -147,9 +148,8 @@ static void
 icapLookupDnsResults(const ipcache_addrs *ia, const Dns::LookupDetails &, void *data)
 {
     Adaptation::Icap::Xaction *xa = static_cast<Adaptation::Icap::Xaction *>(data);
-    /// TODO: refactor with CallJobHere1, passing either std::optional (after upgrading to C++17)
-    /// or Optional<Ip::Address> (when it can take non-trivial types)
-    xa->dnsLookupDone(ia);
+    const auto &addr = ia ? Optional<Ip::Address>(ia->current()) : Optional<Ip::Address>();
+    CallJobHere1(93, 5, CbcPointer<Adaptation::Icap::Xaction>(xa), Adaptation::Icap::Xaction, dnsLookupDone, addr);
```

Comment:
Please remind me, why are we doing this refactoring in _this_ PR? How is the direct call in the official code related to the PR scope?

Reply:
IIRC we [decided](https://measurementfactory.atlassian.net/browse/SQUID-734?focusedCommentId=46600&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-46600) to adjust this case because Optional can now support non-trivial `Ip::Address`.

Reply:
Now that the official Optional supports non-trivial types, please post a dedicated official PR with this small (and out of this PR scope) improvement. If others insist on using a non-reference type for `addr` here, I do not think we should resist, even though the current code might avoid an Ip::Address copy.